beacon-chain/endtoend/endtoend_test.go

```diff
@@ -0,0 +1,514 @@
+package endtoend
+
+import (
+	"bufio"
+	"bytes"
+	"context"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"math/big"
+	"net/http"
+	"os"
+	"os/exec"
+	"path"
+	"strconv"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/bazelbuild/rules_go/go/tools/bazel"
+	"github.com/ethereum/go-ethereum/accounts/abi/bind"
+	"github.com/ethereum/go-ethereum/accounts/keystore"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/core/types"
+	"github.com/ethereum/go-ethereum/ethclient"
+	"github.com/ethereum/go-ethereum/rpc"
+	"github.com/pborman/uuid"
+	ev "github.com/prysmaticlabs/prysm/beacon-chain/endtoend/evaluators"
+	contracts "github.com/prysmaticlabs/prysm/contracts/deposit-contract"
+	"github.com/prysmaticlabs/prysm/proto/eth/v1alpha1"
+	"github.com/prysmaticlabs/prysm/shared/params"
+	"github.com/prysmaticlabs/prysm/shared/testutil"
+	"google.golang.org/grpc"
+)
+
+type end2EndConfig struct {
+	minimalConfig  bool
+	tmpPath        string
+	epochsToRun    uint64
+	numValidators  uint64
+	numBeaconNodes uint64
+	contractAddr   common.Address
+	evaluators     []ev.Evaluator
+}
+
+type beaconNodeInfo struct {
+	processID   int
+	datadir     string
+	rpcPort     uint64
+	monitorPort uint64
+	grpcPort    uint64
+	multiAddr   string
+}
+
+type validatorClientInfo struct {
+	processID   int
+	monitorPort uint64
+}
+
+func runEndToEndTest(t *testing.T, config *end2EndConfig) {
+	tmpPath := path.Join("/tmp/e2e/", uuid.NewRandom().String()[:18])
+	if err := os.MkdirAll(tmpPath, os.ModePerm); err != nil {
+		t.Fatal(err)
+	}
+	config.tmpPath = tmpPath
+	fmt.Printf("Test Path: %s\n", tmpPath)
```

Comment:
```suggestion
```

Reply:
This one is okay to log I think (I'll change to a t.Log) because it's hard to know which path the test occurs in otherwise to read the log files.